File: Uppercase/UppercaseClient.java
import java.io.*;
import java.net.*;
import java.util.Scanner;

public class UppercaseClient {
    public static void main(String[] args) {
        try {
            Scanner scanner = new Scanner(System.in);
            Socket socket = new Socket("localhost", 9090);
            BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
            PrintWriter out = new PrintWriter(socket.getOutputStream(), true);

            // Three line header from server
            System.out.println(in.readLine());
            System.out.println(in.readLine());
            System.out.println(in.readLine());

            while (true) {
                // Send message to server
                System.out.print("send ➜ ");
                String message = scanner.nextLine();
                out.println(message);

                // In case of empty message, server will shutdown
                // and so do the client prompt
                if (message.equals(""))
                    break;

                // Get response
                String response = in.readLine();
                System.out.println(response);
            }

            socket.close();
        } catch (Exception e) {
            System.out.println("Something went wrong!");
        } finally {
            System.out.println("Connection Closed.");
        }
    }
}

File: labs-adv-computer-netowrks/Uppercase/UppercaseClient.py
import socket

def start_client():
    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    client_socket.connect(('localhost', 9090))

    # Receive the header from server 
    print(client_socket.recv(1024).decode('utf-8').strip())
    while True:
        # Send message to server
        message = input("send ➜ ")
        client_socket.sendall((message + '\n').encode('utf-8'))

        # In case of empty message, break and shutdown client
        if not message:
            break

        # Receive response from server
        response = client_socket.recv(1024).decode('utf-8').strip()
        print(f"{response}")

    client_socket.close()
    print("Connection Closed.")

if __name__ == "__main__":
    try:
        start_client()
    except Exception as e:
        print(f"Something went wrong! {e}")
    finally:
        print("Client shutdown.")
